stage1/init/common/pod.go

```diff
@@ -565,6 +565,56 @@ func PodToSystemd(p *stage1commontypes.Pod, interactive bool, flavor string, pri
 	return nil
 }
 
+// evaluateAppMountPath tries to resolve symlinks within the path.
+// It returns the actual relative path for the given path.
+// TODO(yifan): This is a temporary fix for systemd-nspawn not handling symlink mounts well.
+// Could be removed when https://github.com/systemd/systemd/issues/2860 is resolved, and systemd
+// version is bumped.
+func evaluateAppMountPath(appRootfs, path string) (string, error) {
+	link := appRootfs
+
+	paths := strings.Split(path, "/")
+	for i, p := range paths {
+		next := filepath.Join(link, p)
+
+		if !strings.HasPrefix(next, appRootfs) {
+			return "", fmt.Errorf("path escapes app's root: %q", path)
```

Comment:
This would avoid an error if the user uses silly target paths and do what the user probably meant. However, it's not very likely that this would happen and we can merge this PR without it and maybe do it later if we need to.